### PR TITLE
[StructuralMechanicsApplication] Adding factories to strategies

### PR DESCRIPTION
**Pull request by loumalouomega**
**Description**
Adding factories to strategies (not used yet)

**Changelog**
- Adding Factories to strategies
- Adding Parameters constructors
- Adding Parameters methods (Default/Assign)
- Some clean up

### CODE REVIEW

**Comment by philbucher** [comment]
lets hold this until #8654 is solved, I am not sure if there are still some fundamental issues

File: applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp [hunk 1]
```diff
@@ -517,6 +544,47 @@ class EigensolverStrategy
...
+            "eigensolver_settings" : {
+                "solver_type"           : "eigen_eigensystem",
+                "max_iteration"         : 1000,
+                "tolerance"             : 1e-6,
+                "number_of_eigenvalues" : 5,
+                "echo_level"            : 1
+            },
+            "eigensolver_diagonal_values"   : { 
+                "mass_matrix_diagonal_value"      : 0.0,
+                "stiffness_matrix_diagonal_value" : 0.0
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp:572]: those are wrong, check the eigensolver (python)

File: applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp [hunk 2]
```diff
@@ -547,6 +617,39 @@ class EigensolverStrategy
...
+        if (ThisParameters["scheme_settings"].Has("name")) {
+            mpScheme =  SchemeFactoryType().Create(ThisParameters["scheme_settings"]);
+        }
+
+        // Setting up the default builder and solver
+        if (ThisParameters["builder_and_solver_settings"].Has("name")) {
+            const std::string& r_name = ThisParameters["builder_and_solver_settings"]["name"].GetString();
+            if (KratosComponents<TBuilderAndSolverType>::Has( r_name )) {
+                // Defining the linear solver
+                auto p_linear_solver = LinearSolverFactoryType().Create(ThisParameters["linear_solver_settings"]);
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_strategies/custom_strategies/eigensolver_strategy.hpp:644]: this is wrong, an eigensolver is needed here, and this factory doesn't exist yet in cpp

**Review by philbucher — CHANGES_REQUESTED** [review]
please separate the new constructors from the adding of factories

there are already some problems and I don't want to introduce more

sorry :/

**Comment by pooyan-dadvand** [comment]
@KratosMultiphysics/technical-committee thinks that we should close this PR and use the new registry mechanism to register the entities and then create the factories based on the registered entities

**Comment by loumalouomega** [comment]
> @KratosMultiphysics/technical-committee thinks that we should close this PR and use the new registry mechanism to register the entities and then create the factories based on the registered entities

Any example how to do that?

**Comment by loumalouomega** [comment]
Should I close this or adapt it to new registry? @rubenzorrilla

**Comment by rubenzorrilla** [comment]
> Should I close this or adapt it to new registry? @rubenzorrilla

We should definitively use the new registry for such purposes. In this particular case, maybe it's easier to close this one and create a new "registry-based" branch. This is up to you.

**Comment by loumalouomega** [comment]
> > Should I close this or adapt it to new registry? @rubenzorrilla
> 
> We should definitively use the new registry for such purposes. In this particular case, maybe it's easier to close this one and create a new "registry-based" branch. This is up to you.
> 
> EDIT: In any case, and considering that we are to change the strategies, I'd leave this in standby for a while...

In any case I think https://github.com/KratosMultiphysics/Kratos/pull/8672 is still valid

**Pull request closed without merging**